synapse/models/inet.py

```diff
@@ -411,6 +410,45 @@ def _revModl201708141516(self):
         if 'syn:type:fields' not in node[1]:
             self.core.addRows([(node[0], 'syn:type:fields', 'url=inet:url,file=file:bytes', s_common.now())])
 
+    @modelrev('inet', 201708172326)
+    def _revModl201708141516(self):
+
+        props = [
+            ('user',
+             {'ptype': 'inet:netuser', 'doc': 'The netuser associated with the logon event.', 'ro': 1,
+              'form': 'inet:netlogon'}),
+            ('time',
+             {'ptype': 'time', 'doc': 'The time the netuser logged into the service', 'ro': 1,
+              'form': 'inet:netlogon'}),
+            ('status', {'ptype': 'bool', 'ro': 1,
+                        'doc': 'The status of the logon action, denoting if it was successful or not.',
+                        'form': 'inet:netlogon'}),
+            ('ipv4', {'ptype': 'inet:ipv4', 'doc': 'The source IPv4 address of the logon.', 'form': 'inet:netlogon'}),
+            ('ipv6', {'ptype': 'inet:ipv6', 'doc': 'The source IPv6 address of the logon.', 'form': 'inet:netlogon'}),
+            ('logout', {'ptype': 'time', 'doc': 'The time the netuser logged out of the service.',
+                        'form': 'inet:netlogon'})
+        ]
+
+        ttufo = self.core.formTufoByProp('syn:type', 'inet:netlogon',
+                                         **{'subof': 'comp',
+                                            'fields': 'user,inet:netuser|time,time|status,bool',
```

Comment:
I would remove 'status' from the comp type as I don't think it contributes meaningfully to the "uniqueness" of a login event. (There are also many cases where we won't have logon status, so even if it stayed in the comp it would need to be optional.) 

I think the key elements for a logon to an Internet-based service (as opposed to say a lateral movement logon) are the inet:netuser, time, and source (ipv4 or ipv6). Full uniqueness may be dependent on the granularity of the timestamp (you could conceivably have two logon attempts from the same source within one minute...possibly during one second...probably not during the same millisecond). Don't think this "time resolution" difficulty is unique to this event type (originally came up with proposed changes to inet:netpost) but not sure we have a good solution. Then again it may be OK to fold multiple netlogon events together if we have insufficient resolution on the timestamps.

If there are cases when we don't think we'd have the source IP, that could be an optional part of the comp. Thoughts?

Reply:
will we even always know the source ipv(4|6)?  If so, do we want to make those as optional?
That would make the required data:

- inet:netuser
- time

and optional data
- source ip
- status (successful or not)

?